addons/stock_account/models/product.py

```diff
@@ -38,12 +38,12 @@ def write(self, vals):
 
                 # Empty out the stock with the current cost method.
                 description = _(
-                    "Due to a change of product category (from %s to %s), the costing method has changed for product template %s: from %s to %s.",
-                    product_template.categ_id.display_name,
-                    new_product_category.display_name,
-                    product_template.display_name,
-                    product_template.cost_method,
-                    new_product_category.property_cost_method)
+                    "Due to a change of product category (from %(old_category)s to %(new_category)s), the costing method has changed for product template %(product_template)s: from %(old_method)s to %(new_method)s.",
```

Comment:
```suggestion
                    "Due to a change of product category (from %(old_category)s to %(new_category)s), the costing method has changed for product %(product)s: from %(old_method)s to %(new_method)s.",
```
Because `product.template` is labelled "Product", while `product.product` is labelled "Product Variant".